lib/perl/Genome/InstrumentData/AlignmentResult/Merged.pm

```diff
@@ -219,24 +226,70 @@ sub create {
         for my $alignment (@alignments) {
             $alignment->add_user(user => $self, label => 'uses');
         }
-
-        return 1;
-    };
-    if (my $error = $@) {
-        $tx->rollback();
-        die $error;
-    } elsif ($rv ne 1) {
+    }
+    catch {
         $tx->rollback();
-        $self->error_message('Unexpected return value: ' . $rv);
-        die $self->error_message;
-    } else {
-        $tx->commit();
+        die $self->error_message('Merge failed due to error: ' . $_);
     }
+    finally {
+        for my $allocation (@temp_allocations) {
+            $allocation->delete;
+        }
+        unless ($_) {
+            $tx->commit();
```

Comment:
I think the ultimate answer here is that we can do the `finally` blocks in either order, as long as they are in two separate blocks. In this way, deleting the allocations and doing the transaction commit will both have a chance to succeed independently of each other.